include/ignition/gui/MainWindow.hh

```diff
@@ -484,6 +503,9 @@ namespace ignition
       /// \brief Show the plugins menu
       bool showPluginMenu{true};
 
+      /// \brief Show the confirmation dialog on exit
+      bool showDialogOnExit{false};
```

Comment:
New publicly visible data members break ABI, to avoid this you can add it to the [`MainWindowPrivate` class](https://github.com/ignitionrobotics/ign-gui/blob/890a5722640c539068b4c658a8723a74930a9cd6/src/MainWindow.cc#L33) in `MainWindow.cc` and access it using `this->dataPtr`

Reply:
Oh my bad, you are right. Thanks for the suggestion.

Check 50ac123:
In order to avoid breaking ABI I added that member variable to `MainWindowPrivate` class and it was removed from `WindowsConfig`. This change made me move one level up the parsing of the `dialog_on_exit` XML node (to `Application.cc`). Besides, the `dialog_on_exit` XML attribute isn't part of `window` XML node anymore.
It ended up not being as tidy as before. PTAL